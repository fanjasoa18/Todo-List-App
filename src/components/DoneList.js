import React from "react";

function DoneList({done}) {
    
    return(
        <div className="card">
           {done.map((todo) => (
                <li className="list-item" key={todo.id}>
                    <input 
                        type="text" 
                        value={todo.title} 
                        className="list-done" 
                        onChange={(event) => event.preventDefault()} 
                    />
                </li>
           ))} 
        </div>
    )
}

export default DoneList;