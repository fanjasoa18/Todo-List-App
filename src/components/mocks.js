export const todoListMocks = [
    {
        id: 1,
        title: 'banana'
    },
    {
        id: 2,
        title: 'pineapple'
    }
]