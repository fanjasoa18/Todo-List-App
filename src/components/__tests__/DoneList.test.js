import React from 'react';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import {create} from 'react-test-renderer';
import DoneList from '../DoneList';
import {todoListMocks as Mocks} from '../mocks';
import "@testing-library/jest-dom/extend-expect";


describe('Done', () => {
    it("renders correctly", () => {
        const {getByTestId} = render(<DoneList done={Mocks}/>);

        const doneElement = getByTestId("done-test");
        expect(doneElement).toBeInTheDocument();
    });


    test("snapshot renders correctly", () => {
        const renderer = create(<DoneList done={Mocks}/>)
        const json = renderer.toJSON();

        expect(json).toMatchSnapshot();
    });


    it("maps done item", () => {
        render(<DoneList done={Mocks}/>);
        const doneElementS = screen.getAllByTestId("done-list");
        
        expect(doneElementS.length).toEqual(Mocks.length);
        
    });
})