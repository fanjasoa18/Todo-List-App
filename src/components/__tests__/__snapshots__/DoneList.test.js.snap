// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Done snapshot renders correctly 1`] = `
<div
  className="card"
  data-testid="done-test"
>
  <li
    className="list-item"
    data-testid="done-list"
  >
    <input
      className="list-done"
      onChange={[Function]}
      type="text"
      value="banana"
    />
  </li>
  <li
    className="list-item"
    data-testid="done-list"
  >
    <input
      className="list-done"
      onChange={[Function]}
      type="text"
      value="pineapple"
    />
  </li>
</div>
`;